### PR TITLE
Add telemetry sending to collector

### PR DESCRIPTION
**Pull request by gokhan721**

### CODE REVIEW

File: src/logger.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import * as core from '@actions/core'
+
+const LOG_HEADER: string = '[Workflow Telemetry]'
```
Comment (serkan-ozal) [inline src/logger.ts:3]: `Workflow Telemetry` => `Foresight Workflow Kit`

File: src/procTraceParser.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    const completedCommands: CompletedCommand[] = []
+
+    for await (let line of rl) {
+        line = line.trim()
+        if (!line || !line.length) {
+            continue
+        }
+        try {
+            const event = JSON.parse(line)
+            logger.debug(`Parsing trace process event: ${line}`)
```
Comment (serkan-ozal) [inline src/procTraceParser.ts:56]:
```js
if (logger.isDebugEnabled()) {
    logger.debug(`Parsing trace process event: ${line}`)
}
```
might be better

File: src/procTraceParser.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                }
+
+                // Then complete the actual command
+                if (activeCommandCompleted && activeCommand.duration > minDuration) {
+                    completedCommands.push(activeCommand)
+                }
+
+                exitCommandCount++
+            } else {
+                logger.debug(`Unknown trace process event: ${line}`)
```
Comment (serkan-ozal) [inline src/procTraceParser.ts:109]:
```js
if (logger.isDebugEnabled()) {
    logger.debug(`Unknown trace process event: ${line}`)
}
```
might be better

File: src/utils.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+          ciTelemetryData,
+          {
+            headers: {
+              'Content-type': 'application/json; charset=utf-8',
+              'Authorization': `ApiKey ${core.getInput("api_key")}`
+            },
+          },
+        );
+    
+        logger.debug(JSON.stringify(data, null, 4));
```
Comment (serkan-ozal) [inline src/utils.ts:91]:
```js
if (logger.isDebugEnabled()) {
    logger.debug(JSON.stringify(data, null, 4));
}
```

File: src/utils.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        metaData: getMetaData(),
+        telemetryData: telemetryData
+    }
+}
+
+export async function sendData (url :string, ciTelemetryData: CITelemetryData)
+{
+    logger.debug(`Send data url: ${url} ${core.getInput("api_key")}`);
+    try {
+        const { data } = await axios.post(
```
Comment (serkan-ozal) [inline src/utils.ts:80]: Better to check response status code and log error if status code is not `200`

File: src/utils.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+export function createCITelemetryData(telemetryData: ProcessTelemetryDatum): CITelemetryData {
+    return {
+        metaData: getMetaData(),
+        telemetryData: telemetryData
+    }
+}
+
+export async function sendData (url :string, ciTelemetryData: CITelemetryData)
+{
+    logger.debug(`Send data url: ${url} ${core.getInput("api_key")}`);
```
Comment (serkan-ozal) [inline src/utils.ts:78]:
```js
logger.debug(`Sending data (api key=${core.getInput("api_key")}) to url: ${url}`);
```

File: src/utils.ts (hunk 5 above)
Comment (gokhan721) [reply src/utils.ts:80]: Axios automatically throws error w.r.t status code except successful responses and we check it in catch scope with axios.isAxiosError(error).

**Review by serkan-ozal — APPROVED** [review]

**Pull request merged**